src/orchestrator/tests/test_scrap_and_learn.py

```diff
@@ -43,6 +45,28 @@ def classify(doc_content):
             raise ValueError(doc_content)
 
 
+class MockUrlUnicityChecker(object):
+
+    def __init__(self):
+        self.urls_set = set()
+        self.is_unique_count = 0
+        self.saved_count = 0
+
+    def is_unique(self, url):
+        self.is_unique_count += 1
+
+        prev_len = len(self.urls_set)
+
+        self.urls_set.add(url)
+
+        new_len = len(self.urls_set)
+
+        return prev_len != new_len
+
```

Comment:
I suppose the trick on the length is to prevent to make too access two times the set at the expense of readability. It's good for the standard implementation but in mock implementation we could to it the intuitive/direct way to best express the intent, i mean:

if url in urls_set:
    return false
else
    self.urls_set.add(url)
    return true